### ISSUE TITLE
Cannot login with ionic upload

### ISSUE THREAD

**Issue by g-liu** [driftyco/ionic-cli#505]
When I do `ionic upload`, and login with my user/pass, I get the following error message:

```
$ ionic upload
No previous login existed. Attempting to log in now.

To continue, please login to your Ionic account.
Don't have one? Create a one at: https://apps.ionic.io/signup

Email: [redacted]
Password: 
Error logging in



Email or Password incorrect. Please visit https://apps.ionic.io for help. (CLI v1.5.5)

Your system information:

Cordova CLI: 5.1.1
Ionic Version: 1.0.0
Ionic CLI Version: 1.5.5
Ionic App Lib Version: 0.2.2
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Mac OS X Yosemite
Node Version: v0.10.39
Xcode version: Xcode 6.3.2 Build version 6D2105 



```

However I am able to log on to the web interface with the same email and password.

I read the guide at http://ionicframework.com/docs/cli/uploading_viewing.html and it says

<blockquote>
You will be prompted for your username and password (the one you received when you signed up). Once this is done, your app will be uploaded to us.
</blockquote>

But I did not receive a password when I signed up!

I have tried resetting my password and un/re- installing ionic, as well as logging in from older versions of the Ionic CLI. However, none have been successful.

**Comment by g-liu** [comment]
Okay. As strange as this seems, I *downgraded* to version 1.4 with the following command:

`npm install -g ionic@1.4`

Then I ran `ionic login`, then `ionic upload`, and it worked!

**Comment by ericb** [comment]
Made an update to the Ionic.io platform that fixed this.

**Comment by mautematico** [comment]
this is again an issue on `ionic    2.0.0-beta.24`

**Comment by prijindal** [comment]
I think we should open a new issue for ionic 2 cli as login is again not working on them

**Comment by Patrax** [comment]
Log in when doing `ionic io init` isn't working on `ionic 1.7.14` either.

**Comment by wgbn** [comment]
i have same problem in ionic beta.25

**Comment by eshansingh** [comment]
@g-liu Please reopen this issue! This is still a problem on `ionic 2.00-beta.25`

**Comment by JayakumarS** [comment]
I am using 1.7.14, me also having the same problem. I couldn't login through command prompt.

**Comment by patrickhofer** [comment]
+1 please reopen this issue! It's still a problem with ionic 2.00-beta.25...

**Comment by g-liu** [comment]
@eshansingh Unfortunately I cannot reopen this issue. Will have to wait for a collaborator to do that :(

**Comment by xiniux** [comment]
ionic upload, dont working on ionic 1.7.14 :(

**Comment by eshansingh** [comment]
@g-liu See #994 though

**Comment by xiniux** [comment]
ok thanks

2016-05-28 1:07 GMT-04:00 Eshan Singh <notifications@github.com>:

**Comment by SanjayYadav90** [comment]
I, also have same issue,
by using same email and password i can get login on **web (https://apps.ionic.io/apps)**
but
when i use in terminal 
`$ ionic io init`

**it gives erro as-**

**_Email or Password incorrect. Please visit https://apps.ionic.io for help.
Initializing app with ionic.io....
Error initializing app: TypeError: Cannot read property 'map' of undefined_**

**Version I have -**
`$ ionic -v`
2.0.0-beta.31

How to fix it out?

**Comment by makdas** [comment]
so many people with this issue wow and no resolution... all we can do is sit and wait? all production on halt cant share any builds

this is my console log with login details
$ ionic upload
No previous login existed. Attempting to log in now.

To continue, please login to your Ionic account.
Don't have one? Create a one at: https://apps.ionic.io/signup

Email: info@digitalbeyond.co.za
Password: githubpass
Error logging in

Email or Password incorrect. Please visit https://apps.ionic.io for help. (CLI v1.7.16)

Your system information:
Cordova CLI: 6.3.0
Gulp version:  CLI version 3.9.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 1.3.1
Ionic CLI Version: 1.7.16
Ionic App Lib Version: 0.7.3
OS:
Node Version: v4.4.1

**Comment by bhupalasuresh** [comment]
ionic upload
No previous login existed. Attempting to log in now.

To continue, please login to your Ionic account.
Don't have one? Create one at: https://apps.ionic.io/signup

Email: 
Password: 

Email or Password incorrect. Please visit https://apps.ionic.io for help.
Uploading app....
An error occurred uploading the build: TypeError: Cannot read property 'map' of undefined

ypeError: Cannot read property 'map' of undefined
    at Object.getDirectUploadKey (/usr/local/lib/node_modules/ionic/node_modules/ionic-app-lib/lib/upload.js:262:18)
    at /usr/local/lib/node_modules/ionic/node_modules/ionic-app-lib/lib/upload.js:40:21
    at _fulfilled (/usr/local/lib/node_modules/ionic/node_modules/q/q.js:787:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/ionic/node_modules/q/q.js:816:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/ionic/node_modules/q/q.js:749:13)
    at /usr/local/lib/node_modules/ionic/node_modules/q/q.js:509:49
    at flush (/usr/local/lib/node_modules/ionic/node_modules/q/q.js:108:17)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)


Cannot read property 'map' of undefined (CLI v2.0.0)

Your system information:

Cordova CLI: 6.3.1
Ionic Framework Version: 1.3.1
Ionic CLI Version: 2.0.0
Ionic App Lib Version: 2.0.0
ios-deploy version: 1.8.6 
ios-sim version: 5.0.8 
OS: Mac OS X El Capitan
Node Version: v6.3.1
Xcode version: Xcode 7.1 Build version 7B91b

i am getting this error

**Comment by makdas** [comment]
Solution for me was, I used a different ISP. I connected my computer with my phone internet and then the login worked.

**Comment by cassiodourado1** [comment]
@makdas  woking for me, using upload by 3G network,
Thank you!

**Comment by CHOMNANP** [comment]
I had the same problem, and thank for the solution :D
Upload success with my 3G network.
What kind of issue is that!!!

**Comment by lluisnieto** [comment]
Horrible. Tried 2, 3 and 4g on my phone and dsl. Nothing works. I'm about to jump through the window.

**Comment by nealhamilton** [comment]
@lluisnieto  No dice on wifi, or 4g for me either.  ionic -v 3.18.0 cordova -v 7.1.0
